activity_checker.rb

```diff
@@ -50,14 +50,18 @@ def check_activity
         else
           days_without_diff_update = TimeDifference.between(db_pr[:diff_updated], current_time).in_days.to_i
           if days_without_diff_update >= @conf[:timeout]
-            @conf[:recipients].each do |receiver|
-              create_slack_notification_on_outdated_pr(repo, db_pr, days_without_diff_update, receiver)
-            end
-            login = db_pr[:author]
-            user = @controller.get_user_by_login(login)
-            if user and user[:enable]
-              recipient = user[:slack_id]
-              create_slack_notification_on_outdated_pr(repo, db_pr, days_without_diff_update, recipient, login)
+                
+            if pr_need_notification?(db_pr, current_time, days_without_diff_update)
+              @controller.update_pr_notified_at(db_pr[:pr_id], current_time)
+              @conf[:recipients].each do |receiver|
+                create_slack_notification_on_outdated_pr(repo, db_pr, days_without_diff_update, receiver)
+              end
+              login = db_pr[:author]
+              user = @controller.get_user_by_login(login)
+              if user and user[:enable]
+                recipient = user[:slack_id]
+                create_slack_notification_on_outdated_pr(repo, db_pr, days_without_diff_update, recipient, login)
```

Comment:
there will be duplicate messages sent to users that are both recipients and authors of the pull request